dmoj/problem.py

```diff
@@ -104,7 +104,7 @@ class _TestCase:
                 setattr(groups[batch][case], filetype, testcase_file)
 
         test_cases = []
-        for batch_or_case_id in sorted(groups.keys()):
+        for batch_or_case_id in sorted(groups.keys(), key=lambda id : (isinstance(id, int), id)):
```

Comment:
I'm surprised this passed linting...

```suggestion
        for batch_or_case_id in sorted(groups.keys(), key=lambda id: (isinstance(id, int), id)):
```

otherwise LGTM